analyzers/tests/SonarAnalyzer.UnitTest/TestCases/PreferGuidEmpty.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+
+class Compliant
+{
+    Guid? NullableField; // Compliant
+    readonly Guid? ReadonlyNullableField; // Compliant
+    Guid Property { get; set; } // Compliant
+
+    void Empty()
+    {
+        var empty = Guid.Empty; // Compliant
+    }
+
+    void NotEmpty()
+    {
+        var rnd = Guid.NewGuid();// Compliant
+        var bytes = new Guid(new byte[0]); // Compliant
+        var str = new Guid("FA97FFE7-532C-4015-8698-49D8CE4126F4"); // Compliant
+    }
+
+    void NullableDefault()
+    {
+        Guid? nullable = default; // Compliant, not equivalent to Guid.Empty.
+        var nullableOfT = default(Guid?); // Compliant
+        var instance = new NullableGuidClass(default); // Compliant
+        instance.Method(default); // Compliant
+    }
+
+    void NotInitiated(string str)
+    {
+        Guid parsed; // Compliant
+        Guid.TryParse(str, out parsed);
+    }
+
+    void OptionalParameter(Guid id = default) { } // Compliant, default has to be a run-time constant
+
+    void NotGuid()
+    {
+        int integer = default;
+        var date = default(DateTime);
+    }
+}
+
+class NonCompliant
+{
+    Guid Field; // FN
+    readonly Guid ReadonlyField; // FN
+    Guid Property { get; set; }
+
+    void DefaultCtor()
+    {
+        var ctor = new Guid(); // Noncompliant {{Use 'Guid.NewGuid()' or 'Guid.Empty' or add arguments to this GUID instantiation.}}
+        //         ^^^^^^^^^^
+    }
+
+    void DefaultInintiation()
+    {
+        Guid defaultValue = default; // Noncompliant
+        var defaultOfGuid = default(Guid); // Noncompliant
+        Property = default; // Noncompliant
+        var instance = new GuidClass(default); // Noncompliant
+        instance.Method(default); // Noncompliant
```

Comment:
Can you please rename this variable? It took me a while to understand why it is noncompliant.
```suggestion
        var guidAsParameter = new GuidClass(default); // Noncompliant
        guidAsParameter.Method(default); // Noncompliant
```

Reply:
This one was not addressed, it was done on other places. Please try to find all instances of the GuidClass and use it there. I saw 4 `instance` variables that could be renamed